hazelcast/src/test/java/com/hazelcast/util/scheduler/SecondsBasedEntryTaskSchedulerTest.java

```diff
@@ -32,151 +34,151 @@
     private TaskScheduler executorService = mock(TaskScheduler.class);
 
     @Mock
+    @SuppressWarnings("unchecked")
     private ScheduledEntryProcessor<Integer, Integer> entryProcessor = mock(ScheduledEntryProcessor.class);
 
+    private SecondsBasedEntryTaskScheduler<Integer, Integer> scheduler;
+
     @Before
+    @SuppressWarnings("unchecked")
     public void mockScheduleMethod() {
-        when(executorService.schedule(any(Runnable.class), anyLong(), any(TimeUnit.class))).thenReturn(
-                mock(ScheduledFuture.class));
+        when(executorService.schedule(any(Runnable.class), anyLong(), any(TimeUnit.class)))
+                .thenReturn(mock(ScheduledFuture.class));
     }
 
     @Test
     public void test_scheduleEntry_postpone() {
-        final SecondsBasedEntryTaskScheduler<Integer, Integer> scheduler =
-                new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
+        scheduler = new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
 
-        scheduler.schedule(100, 1, 1);
+        assertTrue(scheduler.schedule(100, 1, 1));
         assertNotNull(scheduler.get(1));
         assertEquals(1, scheduler.size());
     }
 
     @Test
     public void test_rescheduleEntry_postpone() {
-        final SecondsBasedEntryTaskScheduler<Integer, Integer> scheduler =
-                new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
+        scheduler = new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
 
-        scheduler.schedule(100, 1, 1);
-        scheduler.schedule(10000, 1, 1);
+        assertTrue(scheduler.schedule(100, 1, 1));
+        assertTrue(scheduler.schedule(10000, 1, 1));
         assertNotNull(scheduler.get(1));
         assertEquals(1, scheduler.size());
     }
 
     @Test
-    public void test_dontRescheduleEntryWithinSameSecond_postpone() {
-        System.setProperty(ClockProperties.HAZELCAST_CLOCK_IMPL, StaticClock.class.getName());
-        try {
-            final SecondsBasedEntryTaskScheduler<Integer, Integer> scheduler =
-                    new SecondsBasedEntryTaskScheduler<Integer, Integer>(executorService, entryProcessor, POSTPONE);
-
-            scheduler.schedule(0, 1, 1);
-            scheduler.schedule(0, 1, 1);
-            assertNotNull(scheduler.get(1));
-            assertEquals(1, scheduler.size());
-        } finally {
-            System.clearProperty(ClockProperties.HAZELCAST_CLOCK_IMPL);
+    public void test_doNotRescheduleEntryWithinSameSecond_postpone() {
```

Comment:
from what I can see, the second assertion will be ignored if the second jumps when running the test. Wouldn't it be better in some kind of loop? i.e. to try doing the assertion until the start/stop seconds are equal.